hdac-io/CasperLabs
# (WIP) Introduction of inflation functionality

### Update
Added features:
```rust
pub fn write_genesis_total_supply(genesis_total_supply: &U512);
pub fn distribute();
pub fn claim_commission(validator: &PublicKey);
pub fn claim_reward(user: &PublicKey);
pub fn step_claim();
```
And its caller:
```rust
pub fn step();
```
@jhyeom26 Currently blocked your implementation in `step()` and will enable with test in next PR.

### How to test
Check the test function:
`./execution-engine/engine-tests/src/test/hdac_system_contracts/proof_of_stake/economy.rs`
Run test: `cargo test hdac_system_contract -- --ignored --nocapture`

**Test scenario:**
1. Write genesis total supply
1. One block passed and distribute
1. Delegate
1. One block passed and distribute
1. Try to claim commission
1. Try to claim reward
1. Send reward token to claimer
